Src/BootCamp.Chapter/TextTable.cs

```diff
@@ -0,0 +1,173 @@
+        {
+            var messageLine = new string[numOfMessageLines];
+            var messageCount = 0;
+            var beginPos = 0;
+            var endPos = 0;
+
+            if (numOfMessageLines == 0 && message.Length > 0)
+            {
+                messageLine[0] = message;
+            }
+
+            for (var i = 0; i < message.Length - 1; i++)
+            {
+                if (message[i] == '\r' && message[i + 1] == '\n')
+                {
+                    endPos = i - 1;
+                    messageLine[messageCount++] = message.Substring(beginPos, endPos - beginPos + 1);
+                    beginPos = i + 2;
+                }
+            }
+
+            if (endPos <= message.Length)
+            {
+                messageLine[messageCount] = message.Substring(beginPos, message.Length - beginPos);
+            }
+
+            return messageLine;
+        }
+
+        private int GetMaxMessageLineLength(string[] messageLine)
+        {
+            var maxLength = 0;
+
+            for (var i = 0; i < messageLine.Length; i++)
+            {
+                if (maxLength < messageLine[i].Length)
+                {
+                    maxLength = messageLine[i].Length;
+                }
+            }
+
+            return maxLength;
+        }
+
+        private string BuildTable(string[] messageLine, int maxLength, int padding)
+        {
+            var tableLine = BuildBorderLine(maxLength, padding, "+", "-");
+            var table = new StringBuilder(tableLine + Environment.NewLine);
+
+            for (var i = 0; i < padding; i++)
+            {
+                tableLine = BuildBorderLine(maxLength, padding, "|", " ");
+                table.Append(tableLine + Environment.NewLine);
+            }
+
+            for (var i = 0; i < messageLine.Length; i++)
+            {
+                tableLine = BuildMessageLine(maxLength, padding, "|", messageLine[i]);              
+                table.Append(tableLine + Environment.NewLine);
+            }
+
+            for (var i = 0; i < padding; i++)
+            {
+                tableLine = BuildBorderLine(maxLength, padding, "|", " ");
+                table.Append(tableLine + Environment.NewLine);
+            }
+
+            tableLine = BuildBorderLine(maxLength, padding, "+", "-");
+            table.Append(tableLine + Environment.NewLine);
+
+            return table.ToString();
+        }
+
+        private string BuildBorderLine(int maxLength, int padding, string sideBorder, string edgeBorder)
+        {
+            var line = new StringBuilder(sideBorder);
+            for (var i = 0; i < maxLength + 2 * padding; i++)
+            {
+                line.Append(edgeBorder);
+            }
+            line.Append(sideBorder);
+
+            return line.ToString();
+        }
+
+        private string BuildMessageLine(int maxLength, int padding, string sideBorder, string messageLine)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Make 'BuildMessageLine' a static method.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5063788)